go.spec

```diff
@@ -53,20 +53,39 @@ any older distro-provided versions.
 %build
 
 %install
+%if (0%{?suse_version} > 0)
+%{__mkdir_p} %{buildroot}/%{_exec_prefix}/bin
+%{__mkdir_p} %{buildroot}/%{_exec_prefix}/lib64/go/%{_go_rel}
+cp -a bin %{buildroot}/%{_exec_prefix}/lib64/go/%{_go_rel}/
+cp -a pkg %{buildroot}/%{_exec_prefix}/lib64/go/%{_go_rel}/
+cp -a src %{buildroot}/%{_exec_prefix}/lib64/go/%{_go_rel}/
+ln -s %{_exec_prefix}/lib64/go/%{_go_rel}/bin/go %{buildroot}/%{_exec_prefix}/bin/go
+ln -s %{_exec_prefix}/lib64/go/%{_go_rel}/bin/gofmt %{buildroot}/%{_exec_prefix}/bin/gofmt
```

Comment:
This approach is the correct path but these link targets will almost assuredly already exist, at least on a system that has a goX.Y package already installed.  I.e.:

```
$ ls -l /usr/bin/go
lrwxrwxrwx. 1 root root 20 Jun  7 16:59 /usr/bin/go -> /etc/alternatives/go
```

IIRC, that `ln -s` will fail if the target exists already.

Using the above information we can see how SUSE handles this:

```
$ rpm -qf /etc/alternatives/go
go1.17-1.17.13-150000.1.45.1.x86_64
go1.20-openssl-1.20.5.2-150000.1.5.1.x86_64
$ rpm -q --scripts go1.17-1.17.13-150000.1.45.1.x86_64
postinstall scriptlet (using /bin/sh):

update-alternatives \
  --install /usr/bin/go go /usr/lib64/go/1.17/bin/go $((20+$(echo 1.17 | cut -d. -f2))) \
  --slave /usr/bin/gofmt gofmt /usr/lib64/go/1.17/bin/gofmt \
  --slave /etc/gdbinit.d/go.gdb go.gdb /usr/lib64/go/1.17/bin/gdbinit.d/go.gdb
postuninstall scriptlet (using /bin/sh):
if [ $1 -eq 0 ] ; then
	update-alternatives --remove go /usr/lib64/go/1.17/bin/go
fi
```

So you could just copy those two scriptlets above into `%post` and `%postun` (guarded by a `(0%{?suse_version} > 0)`) in this spec to avoid the issues with making those links directly.
```

Reply:
Just updated the PR as your suggested. Thank you!